CONTRIBUTING.md

```diff
@@ -1,39 +1,38 @@
 # Contributing
 
-Thanks for your help! Due to ReScript's nature, the contribution setup isn't all straightforward. If something isn't working, please file an issue!
+Thanks for your help! Due to Melange's nature, the contribution setup isn't all straightforward. If something isn't working, please file an issue!
 
 ## Prerequisites
 
-- [NodeJS](https://nodejs.org/)
-- C compiler toolchain (you probably already have it installed)
-- OS: Mac/Linux (ReScript works on Windows, but developing the repo using Windows isn't tested. Contribution welcome!)
+- [NixOS](https://nixos.org/)
 
-## Build
+## Developing
+
+### Nix
+
+The best way to get started is to get a `nix` shell running:
 
 ```sh
-git submodule update --init # we vendor a fork of ocaml and a syntax repo
-node scripts/buildocaml.js # buid the vendored ocaml compiler
-npm install # install some JS tools for testing purposes
-./scripts/ninja.js config # the repo is build with Ninja. Generate the ninja build files
-./scripts/ninja.js build # runs `ninja` under the hood against the generated ninja build files
+# Opens a shell with the necessary environment.
+nix-shell --pure
 ```
 
-Whenever you edit a file, run `./scripts/ninja.js build` to rebuild. Optional watcher to auto-rebuild on file changes: `node scripts/tasks.js`.
+> **Note**: You can also run `nix-shell --command $EDITOR` to get merlin support
 
-In the rare case there you're making changes to the vendored OCaml fork, rebuild the fork with `node scripts/buildocaml.js` then run `./scripts/ninja.js cleanbuild`. `cleanbuild` (aka `clean` + `build`) is necessary since the binary artifacts between versions of compiler may be incompatible.
+Once you're in the shell, you have access to `dune`, `node`, `yarn`, and all the other executables you need to get down to business.
 
-### Troubleshoot Broken Build
+⚠️ Should we maybe list out some common/useful `dune`/`yarn` commands here? I'm personally a big fan of running `dune` in watch mode while I do stuff.
```

Comment:
sounds good. how about the following:

- `dune build` builds the whole project
- `dune runtest` runs the native tests
- `dune exec jscomp/main/js_main.exe` is useful to run the development version of `bsc`